### PR TITLE
Contact Merge Button Error When Clicked Twice

### LINKED ISSUE #5779: Contact Merge Button Error When Clicked Twice
When a customer clicks the Merge button on the Merge Contacts modal twice, they receive the following error:

"Error: Failed to lock all the records for merging. Please retry from start."
Screen Shot 2020-08-21 at 3 10 06 PM

The merge does appear to go through, though, since one of the transactions does complete. This error has come up sporadically for several customers. Some say it occurs when they don't double-click the button, but we can't replicate on our own aside from the double-click (my only theory there is that maybe it's a timing issue with slowness/ lots of related records?).

My request here is to have us disable the Merge button when clicked (like we have in many other areas of our product).

### Steps to Repeat.
1. Create Contact with FirstName = 'Test' and LastName = 'Contact'
2. Create Contact with FirstName = 'Test' and LastName = 'Contact'.
3. Go to the Contact Merge tab and search for "Test Contact."
4. Select the Contacts and click Next.
5. Click the blue Merge button.
6. On the modal that appears, click the red Merge button twice. This should throw the error noted above.

### Root Cause
The error is surfaced in [CON_ContactMerge_CTRL](https://github.com/SalesforceFoundation/NPSP/blob/master/src/classes/CON_ContactMerge_CTRL.cls).

We query for the winning/losing Contacts via [a SOQL query with FOR UPDATE](https://github.com/SalesforceFoundation/NPSP/blob/master/src/classes/CON_ContactMerge_CTRL.cls#L985-L988), to lock the records:

```
List<Contact> lock = [
SELECT Id
   FROM Contact
   WHERE Id IN :allContacts for update];
```

[The error is thrown during the following block of code because the lock size is not equal to size of allContacts.
](https://github.com/SalesforceFoundation/NPSP/blob/master/src/classes/CON_ContactMerge_CTRL.cls#L990)
```
if (lock == null || lock.size() != allContacts.size()) {
   ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.Error,
      label.npe01.Contact_Merge_Error_Lock_failed + ' ' +
      label.npe01.Contact_Merge_Error_please_retry) );
      return null;
}
```

### PR DESCRIPTION
**Pull request by dfuller114**
# Critical Changes

# Changes

# Issues Closed
Fixes #5779 

# Community Ideas Delivered

# Features Intended for Future Release

# Features for Elevate Customers

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: src/pages/CON_ContactMerge.page [hunk 1]
```diff
@@ -298,16 +321,19 @@
                                                         <apex:inputCheckbox value="{!item.selected}"/>
                                                         <span class="slds-checkbox_faux" id="fauxCBSelect"></span>
                                                         <span class="slds-form-element__label slds-assistive-text">
-                                                            {!$Label.conMergeSelectContact} <apex:outputText value="{!item.contactRecord.Name}"/>
+                                                            {!$Label.conMergeSelectContact} <apex:outputText
```
Comment (npsp-reedestockton) [inline src/pages/CON_ContactMerge.page:324]: New line for <apex:outputText...?

**Review by npsp-reedestockton — CHANGES_REQUESTED** [review]
Looks great.  One minor formatting change request.

**Review by lparrott — APPROVED** [review]
Works great, nicely done!

File: src/pages/CON_ContactMerge.page (hunk 1 above)
Comment (dfuller114) [reply src/pages/CON_ContactMerge.page]: updated!

**Review by npsp-reedestockton — APPROVED** [review]

**Comment by npsp-reedestockton** [comment]
Nice!!!

**Comment by rponti-sforg** [comment]
QE testing is complete for this PR and subsequent WI.  Tests have passed based on WI Description, AC(s), PR comments, and design document

For further testing details, see:
https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008UcphIAC/view
https://salesforce.quip.com/drZ9AvafFjp7

Proceeding with merge of PR.  Will set WI status to 'Pending Release' and then 'Closed' once post-merge builds pass.

**Pull request merged**